tests/func/test_add.py

```diff
@@ -529,37 +531,36 @@ def test_windows_should_add_when_cache_on_different_drive(
     assert ret == 0
 
 
-def test_readding_dir_should_not_unprotect_all(dvc_repo, repo_dir):
-    dvc_repo.cache.local.cache_types = ["symlink"]
-    dvc_repo.cache.local.protected = True
-
-    dvc_repo.add(repo_dir.DATA_DIR)
-    new_file = os.path.join(repo_dir.DATA_DIR, "new_file")
+def test_readding_dir_should_not_unprotect_all(dvc, tmp_dir, mocker):
+    tmp_dir.gen({"dir": {"data": "data"}})
+    dvc.cache.local.cache_types = ["symlink"]
+    dvc.cache.local.protected = True
 
-    repo_dir.create(new_file, "new_content")
+    dvc.add("dir")
+    tmp_dir.gen({"dir": {"new_file": "new_file_content"}})
 
     unprotect_spy = spy(RemoteLOCAL.unprotect)
-    with patch.object(RemoteLOCAL, "unprotect", unprotect_spy):
-        dvc_repo.add(repo_dir.DATA_DIR)
+    mocker.patch.object(RemoteLOCAL, "unprotect", unprotect_spy)
+    dvc.add("dir")
 
     assert not unprotect_spy.mock.called
-    assert System.is_symlink(new_file)
+    assert System.is_symlink(os.path.join("dir", "new_file"))
 
 
-def test_should_not_checkout_when_adding_cached_copy(repo_dir, dvc_repo):
-    dvc_repo.cache.local.cache_types = ["copy"]
+def test_should_not_checkout_when_adding_cached_copy(dvc, tmp_dir, mocker):
+    dvc.cache.local.cache_types = ["copy"]
 
-    dvc_repo.add(repo_dir.FOO)
-    dvc_repo.add(repo_dir.BAR)
-
-    shutil.copy(repo_dir.BAR, repo_dir.FOO)
+    tmp_dir.gen({"foo": "foo", "bar": "bar"})
+    dvc.add("foo")
+    dvc.add("bar")
```

Comment:
Replace 3 lines with:
```suggestion
    tmp_dir.dvc_gen({"foo": "foo", "bar": "bar"})
```